arrow/compute/scalar_compare.go

```diff
@@ -134,4 +136,50 @@ func RegisterScalarComparisons(reg FunctionRegistry) {
 	reg.AddFunction(ltFn, false)
 	lteFn := makeFlippedCompare("less_equal", gteFn, EmptyFuncDoc)
 	reg.AddFunction(lteFn, false)
+
+	isOrNotNullKns := kernels.IsNullKernels()
+	isNullFn := &compareFunction{*NewScalarFunction("is_null", Unary(), EmptyFuncDoc)}
+	if err := isNullFn.AddKernel(isOrNotNullKns[0]); err != nil {
+		panic(err)
+	}
+
+	isNotNullFn := &compareFunction{*NewScalarFunction("is_not_null", Unary(), EmptyFuncDoc)}
+	if err := isNotNullFn.AddKernel(isOrNotNullKns[1]); err != nil {
+		panic(err)
+	}
+
+	reg.AddFunction(isNullFn, false)
+	reg.AddFunction(isNotNullFn, false)
+
+	reg.AddFunction(NewMetaFunction("is_nan", Unary(), EmptyFuncDoc,
+		func(ctx context.Context, opts FunctionOptions, args ...Datum) (Datum, error) {
+			switch args[0].Kind() {
```

Comment:
Nit: could you reoder the type switch to be outside the kind switch? Then we have one `return CallFunction(ctx, "not_equal", nil, arg, arg)` for all kinds of floats and we either return scalar false or broadcast it for other types

Reply:
reordered and updated